query/src/main/scala/hmda/query/repository/filing/FilingComponent.scala

```diff
@@ -296,28 +299,79 @@ trait FilingComponent { this: DbConfiguration =>
+    def home_improve = column[Int]("home_improve")
+    def refinance = column[Int]("refinance")
+
+    override def * = (
+      msa,
+      total_lars,
+      total_amount,
+      conv,
+      fha,
+      va,
+      fsa,
+      oneToFourFamily,
+      manuf_home,
+      multi_family,
+      home_purchase,
+      home_improve,
+      refinance
+    ) <> (Msa.tupled, Msa.unapply)
   }
 
-  class LarTotalRepository(val config: DatabaseConfig[JdbcProfile]) extends Repository[LarTotalTable, Int] {
-    val table = TableQuery[LarTotalTable]
-    def getId(table: LarTotalTable) = table.id
+  class LarTotalRepository(val config: DatabaseConfig[JdbcProfile]) extends Repository[LarTotalTable, String] {
+    implicit val system = ActorSystem()
+    implicit val materializer = ActorMaterializer()
 
-    private val createViewSchema = sqlu"""create view lars_total as
-      select count(*) as total, respondent_id, period
-      from lars group by respondent_id, period
+    val configuration = ConfigFactory.load()
+    val queryFetchSize = configuration.getInt("hmda.query.fetch.size")
+
+    val table = TableQuery[LarTotalTable]
+    def getId(table: LarTotalTable) = table.msa
+
+    private def createViewSchema(period: String) = {
+      sqlu"""create view lars_total as
+        select msa, count(*) as total_lars, sum(amount) as total_amount,
+        count(case when loan_type = 1 then 1 else null end) as conv,
+        count(case when loan_type = 2 then 1 else null end) as fha,
+        count(case when loan_type = 3 then 1 else null end) as va,
+        count(case when loan_type = 4 then 1 else null end) as fsa,
+        count(case when property_type = 1 then 1 else null end) as one_to_four_family,
+        count(case when property_type = 2 then 1 else null end) as manuf_home,
+        count(case when property_type = 3 then 1 else null end) as multi_family,
+        count(case when purpose = 1 then 1 else null end) as home_purchase,
+        count(case when purpose = 2 then 1 else null end) as home_improve,
+        count(case when purpose = 3 then 1 else null end) as refinance
+        from lars where period = '#$period' group by msa;
       """
+    }
 
-    def createSchema() = db.run(createViewSchema)
+    def createSchema(period: String) = db.run(createViewSchema(period))
     def dropSchema() = db.run(table.schema.drop)
-    def count(respId: String) = db.run(table.filter(_.respondentId === respId).map(_.total).result.headOption)
 
+    private def getTableStream()(implicit ec: ExecutionContext): DatabasePublisher[Msa] = {
+      val disableAutocommit = SimpleDBIO(_.connection.setAutoCommit(false))
+      val query = table.drop(0)
+      val action = query.result.withStatementParameters(fetchSize = queryFetchSize)
+
+      db.stream(disableAutocommit andThen action)
+    }
+
+    def getMsaSeq()(implicit ec: ExecutionContext): Future[Seq[Msa]] = {
+      Source.fromPublisher(getTableStream()).grouped(1000).runWith(Sink.head)
```

Comment:
It might make sense to put this value into the configuration file and make it a parameter